### DISCUSSION TITLE
How to set up WebRTC blocking?

### DISCUSSION THREAD

**Discussion by galaczi** [ZFC-Digital/puppeteer-real-browser#195] [Q&A]
First of all thanks for this awesome tool! However I am being detected by WebRTC. I have seen a couple issues about it, but nothing I could use to solve it. Any tips?

**Comment by RubberArchind** [comment]
can you explain more what you mean by detected by WebRTC? and what exactly you are doing

**Reply by galaczi** [reply]
I use a proxy. Detection tools open a WebRTC connection which bypasses the proxy, so they can see my real IP, and the fact that I am using a proxy.

https://browserleaks.com/webrtc
https://www.browserscan.net/
https://github.com/zfcsoftware/puppeteer-real-browser/issues/38

**Reply by RubberArchind** [reply]
have you try to use the lastest commit version?

**Reply by RubberArchind** [reply]
and i think this would help https://github.com/puppeteer/puppeteer/issues/4149#issuecomment-550333435

**Reply by galaczi** [reply]
I used the latest commit version. I am able to pass Cloudflare, but the website blocks me if I use a proxy. I tried with residential proxies too. The only thing missing is WebRTC.

Setting those values to undefined is a giveaway that something is not right, and the browser starts to fail Cloudflare checks too.

**Reply by prashant-vr** [reply]
One simple way to prevent webRTC leak is:

Linux: Use `ufw` to block UDP protocol system wide.
Windows: Use `simplewall` or another software to block UDP for the chrome used by `puppeteer-real-browser`

No need to configure webrtc prevention in your code. Since webRTC uses UDP protocol, you are blocking it.

**Comment by prettymaya** [comment]
guys
this method worked %90 but not automaticly (its saying different time. i changed time with code but it didint work and even say you have proxy so what can i do idk but it didint detect to proxy. if anyone solve problem of time, can one explain solution )
write on terminal 
git clone https://github.com/aghorler/WebRTC-Leak-Prevent extensions/webrtc

add this code on your code
const extensionPath = path.join(__dirname, 'extensions', 'webrtc');

args: [
--disable-extensions-except=${extensionPath},
--load-extension=${extensionPath}
],

![Ekran görüntüsü 2024-10-11 000155](https://github.com/user-attachments/assets/8fba1961-11be-4d3c-bdcd-c6f514b0b7e6)
![Ekran görüntüsü 2024-10-11 000206](https://github.com/user-attachments/assets/3a4bff0a-bab6-4278-810a-b9382ef5d73f)
![Ekran görüntüsü 2024-10-11 000426](https://github.com/user-attachments/assets/6911792a-4e59-45fd-b5ac-d1aa72aaebdd)

also https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/228

**Reply by galaczi** [reply]
Hi @barutlift, since Chrome Launcher creates a new profile on every launch, how do you persist the settings of the extension?

**Reply by prettymaya** [reply]
@galaczi i found something like this https://stackoverflow.com/questions/53017490/how-can-i-disable-webrtc-local-ip-leak-with-puppeteer
look at advanced part maybe it works ill try it when i wake up goodn

**Reply by prettymaya** [reply]
@galaczi that worked automatically 
first step 
you can see extentions/webrtc/js/background.js
open background.js and only change them like on the image it will work 
![Ekran görüntüsü 2024-10-12 111713](https://github.com/user-attachments/assets/ba25b232-7f86-43c1-b5a8-1ea4a447653c)

**Reply by prettymaya** [reply]
but if you save profile data, you can only use first enter on chromium so if you will twice use saved data or twice enter same data, you have to change manuel first choose except for udp and than choose udp it will work
![Ekran görüntüsü 2024-10-12 113843](https://github.com/user-attachments/assets/16ce942e-7062-4592-b1e2-fcb68dbae867)

**Reply by galaczi** [reply]
It works, thanks! Turns out it was not WebRTC that got me detected. It's probably something about my proxies. Without proxy I can do whatever I want. With proxies I pass Cloudflare but the website soft blocks me.

**Comment by prashant-vr** [comment]
> First of all thanks for this awesome tool! However I am being detected by WebRTC. I have seen a couple issues about it, but nothing I could use to solve it. Any tips?

Try these flags:
```
args: [`--disable-blink-features=AutomationControlled`,
            '--no-sandbox',
            '--disable-webrtc',
            '--disable-features=WebRtcUdpPortAllocation',
            '--force-webrtc-ip-handling-policy=disable_non_proxied_udp'],
```
Last 3 are meant for webrtc and for this worked without blocking UDP
